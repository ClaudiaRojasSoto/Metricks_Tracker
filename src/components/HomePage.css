.home-page {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  height: 100vh;
}

img {
  width: 45%;
  max-width: 300px;
  height: auto;
}

.continent-grid {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  max-width: 900px;
  margin: 0 auto;
}

.continent-row {
  display: flex;
  justify-content: space-around;
  width: 100%;
  max-width: 600px;
  margin-bottom: 20px;
}
