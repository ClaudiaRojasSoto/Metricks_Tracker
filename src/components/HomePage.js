import React from 'react';
import { Link } from 'react-router-dom';
import world from '../imagesContinents/continents/world.png';

import africa from '../imagesContinents/continents/africa.png';
import america from '../imagesContinents/continents/america.png';
import asia from '../imagesContinents/continents/asia.png';
import australia from '../imagesContinents/continents/australia.png';
import europa from '../imagesContinents/continents/europa.png';
import oceania from '../imagesContinents/continents/oceania.png';

const HomePage = () => {
  const continents = [
    { name: 'africa', image: africa },
    { name: 'america', image: america },
    { name: 'asia', image: asia },
    { name: 'australia', image: australia },
    { name: 'europa', image: europa },
    { name: 'oceania', image: oceania },
  ];

  return (
    <>
      <header>
        <img src={world} alt={world} />
      </header>
      <div className="home">
        {continents.map((continent) => (
          <Link to={`/countries/${continent.name}`} key={continent.name}>
            <img src={continent.image} alt={continent.name} />
            <h2>{continent.name}</h2>
          </Link>
        ))}
      </div>
    </>
  );
};

export default HomePage;
