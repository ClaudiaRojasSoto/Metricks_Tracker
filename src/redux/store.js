import { configureStore } from '@reduxjs/toolkit';
import countriesReducer from './reducers/countriesReducer';

const store = configureStore({
  reducer: {
    countries: countriesReducer,
  },
});

export default store;
