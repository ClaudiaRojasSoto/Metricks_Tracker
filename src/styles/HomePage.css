.home-page {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100vh;
}

.continent-grid {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  max-width: 900px;
  margin: 0 auto;
}

.continent-row {
  display: flex;
  justify-content: space-around;
  width: 100%;
  max-width: 600px;
  margin-bottom: 20px;
}

img {
  width: 45%; /* Ajusta el ancho de las im√°genes para que quepan dos por fila */
  max-width: 300px;
  height: auto;
}
