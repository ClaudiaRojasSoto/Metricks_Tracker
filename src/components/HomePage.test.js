import '@testing-library/jest-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/store';
import HomePage from './HomePage';

describe('HomePage tests', () => {
  let homePage;

  beforeAll(() => {
    homePage = renderer
      .create(
        <Provider store={store}>
          <Router>
            <HomePage />
          </Router>
        </Provider>,
      )
      .toJSON();
  });

  it('Component should render correctly', () => {
    expect(homePage).toBeTruthy();
  });

  it('It should match its snapshot', () => {
    expect(homePage).toMatchSnapshot();
  });
});
