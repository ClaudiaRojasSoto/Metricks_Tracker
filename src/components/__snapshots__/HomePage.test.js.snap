// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage tests It should match its snapshot 1`] = `
<div
  className="home"
>
  <header>
    <img
      alt="world.png"
      className="world_img"
      src="world.png"
    />
    <div
      className="title-world-population"
    >
      <h3>
        World Population
      </h3>
      <p>
        0
      </p>
    </div>
  </header>
  <div
    className="continents-container"
  >
    <div
      className="continents"
    >
      <a
        href="/countries/africa"
        onClick={[Function]}
      >
        <img
          alt="africa"
          src="africa.png"
        />
        <div
          className="title-population"
        >
          <h3>
            africa
          </h3>
          <p />
        </div>
      </a>
    </div>
    <div
      className="continents"
    >
      <a
        href="/countries/americas"
        onClick={[Function]}
      >
        <img
          alt="americas"
          src="america.png"
        />
        <div
          className="title-population"
        >
          <h3>
            americas
          </h3>
          <p />
        </div>
      </a>
    </div>
    <div
      className="continents"
    >
      <a
        href="/countries/asia"
        onClick={[Function]}
      >
        <img
          alt="asia"
          src="asia.png"
        />
        <div
          className="title-population"
        >
          <h3>
            asia
          </h3>
          <p />
        </div>
      </a>
    </div>
    <div
      className="continents"
    >
      <a
        href="/countries/europe"
        onClick={[Function]}
      >
        <img
          alt="europe"
          src="europa.png"
        />
        <div
          className="title-population"
        >
          <h3>
            europe
          </h3>
          <p />
        </div>
      </a>
    </div>
    <div
      className="continents"
    >
      <a
        href="/countries/oceania"
        onClick={[Function]}
      >
        <img
          alt="oceania"
          src="oceania.png"
        />
        <div
          className="title-population"
        >
          <h3>
            oceania
          </h3>
          <p />
        </div>
      </a>
    </div>
  </div>
</div>
`;
